app/controllers/coronavirus_form/check_answers_controller.rb

```diff
@@ -10,24 +10,26 @@ def show
   end
 
   def submit
-    submission_reference = reference_number
-
-    session[:reference_number] = submission_reference
+    session[:reference_number] = reference_number
     FormResponse.create(form_response: session)
 
     reset_session
 
-    redirect_to controller: "coronavirus_form/thank_you", action: "show", reference_number: submission_reference
+    redirect_to controller: "coronavirus_form/thank_you", action: "show", reference_number: reference_number
   end
 
 private
 
   helper_method :items_part_1, :items_part_2
 
+  QUESTIONS = YAML.load_file(Rails.root.join("config/locales/en.yml")).dig("en", "coronavirus_form", "questions").keys
+
   def reference_number
-    timestamp = Time.zone.now.strftime("%Y%m%d-%H%M%S")
-    random_id = SecureRandom.hex(3).upcase
-    "#{timestamp}-#{random_id}"
+    @reference_number ||= begin
```

Comment:
So fragment caching == memoizing any variables that have computational overhead?

Reply:
Yeah, although since the method is already called just once, it won't lead to a performance improvement.